### PR TITLE
Adds Category Bar Chart

### PR DESCRIPTION
**Pull request by dani-moreno**
Adds a new type of chart to our Charts category: **Category Bar**, as we have in [Figma](https://www.figma.com/design/O8578qyVUfgzm6ZkVkBJwW/%E2%8C%9B%EF%B8%8F-Insight-UI-Kit-2.0?node-id=500-43508&t=ZmEsXqhlWOXmyzRJ-4).

https://github.com/user-attachments/assets/baa2c67d-8dd4-4c99-831a-8176e7b3df09

### CODE REVIEW

File: lib/components/Charts/CategoryBarChart/index.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+              const percentage = (value / total) * 100
+              return percentage % 1 === 0
+                ? percentage.toFixed(0)
+                : percentage.toFixed(1)
+            }
+
+            return (
+              <Tooltip key={category.name}>
+                <TooltipTrigger
+                  className="h-full cursor-default overflow-hidden rounded-[2px]"
```
Comment (RamProg) [inline lib/components/Charts/CategoryBarChart/index.tsx:40]: I think we are avoiding the custom values like rounded-[2px], but I understand it's the only way

File: lib/components/Charts/CategoryBarChart/index.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                    tabIndex={0}
+                  >
+                    {category.muted && (
+                      <div
+                        className="absolute inset-0"
+                        style={{
+                          backgroundImage: `repeating-linear-gradient(
+                          135deg,
+                          rgba(255,255,255,0.2),
+                          rgba(255,255,255,0.2) 2px,
```
Comment (RamProg) [inline lib/components/Charts/CategoryBarChart/index.tsx:59]: can we use an existing color so it's compatible with dark mode?

File: lib/components/Charts/CategoryBarChart/index.tsx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                          transparent 6px
+                        )`,
+                        }}
+                      />
+                    )}
+                  </div>
+                </TooltipTrigger>
+                <TooltipContent className="flex items-center gap-1 text-sm">
+                  <div
+                    className="h-2.5 w-2.5 shrink-0 translate-y-[1px] rounded-full"
```
Comment (RamProg) [inline lib/components/Charts/CategoryBarChart/index.tsx:70]: same here about custom values

File: lib/components/Charts/CategoryBarChart/index.tsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+            const color = category.color || autoColor(index)
+
+            return (
+              <div
+                key={category.name}
+                className="flex items-center gap-1"
+                role="listitem"
+              >
+                <div
+                  className="h-2 w-2 shrink-0 translate-y-[1px] rounded-full"
```
Comment (RamProg) [inline lib/components/Charts/CategoryBarChart/index.tsx:100]: same here about custom values

**Review by RamProg — APPROVED** [review]
Nice! I left a few comments to take a look

File: lib/components/Charts/CategoryBarChart/index.tsx (hunk 3 above)
Comment (dani-moreno) [reply lib/components/Charts/CategoryBarChart/index.tsx:70]: Good catch, there's even a Tailwind class that does the same thing, `translate-y-px`, so no need for a custom value! Changing it!

File: lib/components/Charts/CategoryBarChart/index.tsx (hunk 2 above)
Comment (dani-moreno) [reply lib/components/Charts/CategoryBarChart/index.tsx:59]: Umm, this is being displayed on top of the bar, and with the opacity it looks good IMO no matter the theme.
image

**Pull request merged**